src/bosh-dns/dns/config/addresses/fs_loader.go

```diff
@@ -0,0 +1,38 @@
+package addresses
+
+import (
+	"encoding/json"
+	"errors"
+
+	bosherr "github.com/cloudfoundry/bosh-utils/errors"
+	boshsys "github.com/cloudfoundry/bosh-utils/system"
+)
+
+type FSLoader struct {
+	fs boshsys.FileSystem
+}
+
+func (l FSLoader) Load(filename string) (AddressConfigs, error) {
+	fileContents, err := l.fs.ReadFile(filename)
+	if err != nil {
+		return nil, bosherr.WrapError(err, "missing address config file")
+	}
+
+	var addresses AddressConfigs
+	err = json.Unmarshal(fileContents, &addresses)
+	if err != nil {
+		return nil, bosherr.WrapErrorf(err, "addresses config file malformed: %s", filename)
+	}
+
+	for _, c := range addresses {
+		if c.Port == 0 {
```

Comment:
should we be validating address as well, or are we allowing empty string to mean something?

Reply:
we just matched the existing behavior here: https://github.com/cloudfoundry/bosh-dns-release/blob/master/src/bosh-dns/dns/config/config.go

which does not do any validation of the address